File: dotnet/DesignPatterns/CreationalDesignPatterns/BuilderDesignPattern/BuilderDesignPattern.Example1/Report.cs
namespace BuilderDesignPattern.Example1;

public class Report
{
    public string Title { get; set; } = string.Empty;
    public string Content { get; set; } = string.Empty;
    public string? Header { get; set; }
    public string? Footer { get; set; }
    public string? Font { get; set; }


    public override string ToString()
    {
        return $"Report:\nTitle: {Title}\nContent: {Content}\nHeader: {Header}\nFooter: {Footer}\nFont: {Font}";
    }

}

File: nodejs/DesignPatterns/CreationalDesignPatterns/BuilderDesignPattern/BuilderDesignPattern.Example1/src/Report.ts
export class Report {
  title: string = '';
  content: string = '';
  header?: string;
  footer?: string;
  font?: string;

  toString(): string {
    return `Report:\nTitle: ${this.title}\nContent: ${this.content}\nHeader: ${this.header}\nFooter: ${this.footer}\nFont: ${this.font}`;
  }
}
